vault/resource_identity_entity.go

```diff
@@ -67,23 +67,41 @@ func identityEntityResource() *schema.Resource {
 	}
 }
 
-func identityEntityUpdateFields(d *schema.ResourceData, data map[string]interface{}) {
-	if name, ok := d.GetOk("name"); ok {
-		data["name"] = name
-	}
+func identityEntityUpdateFields(d *schema.ResourceData, data map[string]interface{}, create bool) {
+	if create {
+		if name, ok := d.GetOk("name"); ok {
+			data["name"] = name
+		}
 
-	if externalPolicies, ok := d.GetOk("external_policies"); !(ok && externalPolicies.(bool)) {
-		if policies, ok := d.GetOk("policies"); ok {
-			data["policies"] = policies.(*schema.Set).List()
+		if externalPolicies, ok := d.GetOk("external_policies"); !(ok && externalPolicies.(bool)) {
+			if policies, ok := d.GetOk("policies"); ok {
+				data["policies"] = policies.(*schema.Set).List()
+			}
 		}
-	}
 
-	if metadata, ok := d.GetOk("metadata"); ok {
-		data["metadata"] = metadata
-	}
+		if metadata, ok := d.GetOk("metadata"); ok {
+			data["metadata"] = metadata
+		}
 
-	if disabled, ok := d.GetOk("disabled"); ok {
-		data["disabled"] = disabled
+		if disabled, ok := d.GetOk("disabled"); ok {
+			data["disabled"] = disabled
+		}
+	} else {
+		if d.HasChange("name") {
+			data["name"] = d.Get("name")
+		}
+
+		if d.HasChange("policies") {
+			data["policies"] = d.Get("policies").(*schema.Set).List()
+		}
+
+		if d.HasChange("metadata") {
+			data["metadata"] = d.Get("metadata")
+		}
+
+		if d.HasChange("disabled") {
+			data["disabled"] = d.Get("disabled")
+		}
```

Comment:
BTW, it also looks like this can be compacted a bit like this:

```
		if d.HasChanges("name", "policies", "metadata", "disabled") {
			data["name"] = d.Get("name")
			data["policies"] = d.Get("policies").(*schema.Set).List()
			data["metadata"] = d.Get("metadata")
			data["disabled"] = d.Get("disabled")
		}
```

Since it seems that attributes which aren't changed just return their current setting from d.Get()?

Reply:
This is much nicer, thanks! I also needed to add some extra logic because there was an edge case with `external_policies`. A user might enabled external policies, but left policies in the plan. In this situation the policies should be removed from the entity.

Reply:
Nice catch on the `external_policies` case!